src/GroupManager.php

```diff
@@ -282,17 +284,31 @@ public function refresh() {
   }
 
   /**
-   * Creates default roles for the given group type.
+   * Creates per bundle or per group OG roles.
    *
    * @param string $entity_type_id
    *   The entity type ID of the group for which to create default roles.
-   * @param string $bundle_id
-   *   The bundle ID of the group for which to create default roles.
+   * @param string|null $bundle_id
+   *   (Optional) The bundle ID of the group for which to create default roles.
+   *   When this property is set it means the roles will be "global" roles
+   *   that apply to any group of the bundle ID. Defaults to NULL.
+   * @param string|null $group_id
+   *   (Optional) The group ID of the group for which to create default roles.
+   *   When this property is set it means the roles will be "per group" roles
+   *   and apply only to this specific group ID. Defaults to NULL.
+   *
+   * @see \Drupal\og\GroupManager::createPerBundleRoles()
+   * @see \Drupal\og\GroupManager::createPerGroupRoles()
    */
-  protected function createRoles($entity_type_id, $bundle_id) {
+  protected function createRoles($entity_type_id, $bundle_id = NULL, $group_id = NULL) {
```

Comment:
`OgRole` is a config entity, I don't think this supports bundles. If you look at the inherited method `OgRole::bundle()` you see it returns the entity type instead of the bundle. Only content entities support custom bundles out of the box.

But I actually think we can hack in bundle support easily. We just need to ensure that the bundle is stored along with the entity, that we have a `$this->bundle` property that we populate in the constructor, and then we can use it to handle `per_bundle` and `per_group` specific logic.